### DISCUSSION TITLE
Nasa Tlx - displaying multiple sliders on one page plugin?

### DISCUSSION THREAD

**Discussion by jenniekim00** [jspsych/jsPsych#799] [Q&A] [answered]
Has anyone tried recreating the NASA-TLX questionnaire with jspsych? 

I am currently trying out the image-slider-plugin for the rating scales.  However, I haven't figured out how to display multiple rating scales on the same page yet. If you have any ideas, thanks!

P.S. I have found other sources of web-based NASA-TLX online:
- https://www.keithv.com/software/nasatlx/   
- https://github.com/sirctseb/NASA-TLX
I haven't figured out how to integrate it into my current experiment using jspsych. 

My overall goal is to incorporate both a custom image complex span task (in progress) + nasa tlx for my thesis project.

**Comment by vijaymarupudi** [comment, marked as answer]
Looking at the source for this one: https://www.keithv.com/software/nasatlx/nasatlx.html, it would be relatively easy to transform it to a jsPsych plugin.

You can pretty much copy and paste most of the code, just changing a few bits to interact with jsPsych.

https://www.jspsych.org/plugins/creating-a-plugin/ will help!

**Reply by jenniekim00** [reply]
Hi there, thanks for the response! The only issue is figuring out how to 'copy and paste' the html code, i can easily copy and paste the js code and adjust into a plugin, but figuring out how to recode the html into the plugin is a lil hard for me to wrap my head around- the first thing i can think of is recoding with the '.innerHTML' but it's still a lot for my novice brain to fully understand yet aha... I already have an html file formatted with jspsych plugins for my current working memory tasks in progress, so i simply can't copy & paste unfortunately. 

But definitely I will look into it more! If you or anyone knows how to display multiple slider scales on one page with a plugin, that'd be great!! I'm still in the progress of debugging and creating more code, any help will be appreciated :'))

**Reply by vijaymarupudi** [reply]
Here's a working [demo]. I've made some functions global where the html expects them to be global:

```js
    window.buttonPart1 = buttonPart1;
    window.buttonPart2 = buttonPart2;
    window.buttonPair1 = buttonPair1;
    window.buttonPair2 = buttonPair2;
```

and

```js
    window.scaleClick = function scaleClick(index, val) {
```

And instead of displaying the results, I've collected them into an object and called `jsPsych.finishTrial()`.


```js
        calcResults();
        var finalResults = { scales: scale, ratings: results_rating, tallies: results_tally, weights: results_weight }
        jsPsych.finishTrial(finalResults)
```

Here's the code for the entire plugin.

```js
jsPsych.plugins["tlx"] = {
  info: { parameters: {} },
  trial: function (display_element, trial) {
    display_element.innerHTML = `
           <div id="div_part1">
           Task Questionnaire - Part 1 <br />
           <br />
           Click on each scale at the point that best indicates your experience of the task <br />
           <br />
           <div id="scale0"></div>
           <div id="scale1"></div>
           <div id="scale2"></div>
           <div id="scale3"></div>
           <div id="scale4"></div>
           <div id="scale5"></div>
           <br />
           <input class="next" id="next" type="button" value="Continue >>" onclick="buttonPart1();">
           </div>
           <div id="div_part2" style="display:none">
           Task Questionnaire - Part 2 <br />
           <br />
           One each of the following 15 screens, click on the scale title that represents the
           more important contributor to workload for the task <br />
           <br />
           <input class="next" id="next2" type="button" value="Continue >>" onClick="buttonPart2();">
           </div>
           <div id="div_part3" style="display:none">
           Task Questionnaire - Part 2 <br />
           <br />
           Click on the factor that represents the more important contributor to workload for the task<br />
           <br />
           <table>
           <tr>
           <td><input class="pair" id="pair1" type="button" value="" onClick="buttonPair1();"> </td>
           <td class="def"><div id="pair1_def"></div></td>
           </tr>
           <tr>
           <td align="center"> or </td>
           <td></td>
           </tr>
           <tr>
           <td><input class="pair" id="pair2" type="button" value="" onClick="buttonPair2();"></td>
           <td class="def"><div id="pair2_def"></div></td>
           </tr>
           </table>
           </div>
           <div id="div_part4" style="display:none">
           </div>
           `;

    window.buttonPart1 = buttonPart1;
    window.buttonPart2 = buttonPart2;
    window.buttonPair1 = buttonPair1;
    window.buttonPair2 = buttonPair2;

    var scale = new Array();
    var left = new Array();
    var right = new Array();
    var def = new Array();
    var NUM_SCALES = 6;

    scale[0] = "Mental Demand";
    left[0] = "Low";
    right[0] = "High";
    def[0] =
      "How much mental and perceptual activity was required (e.g. thinking, deciding, calculating, remembering, looking, searching, etc)? Was the task easy or demanding, simple or complex, exacting or forgiving?";

    scale[1] = "Physical Demand";
    left[1] = "Low";
    right[1] = "High";
    def[1] =
      "How much physical activity was required (e.g. pushing, pulling, turning, controlling, activating, etc)? Was the task easy or demanding, slow or brisk, slack or strenuous, restful or laborious?";

    scale[2] = "Temporal Demand";
    left[2] = "Low";
    right[2] = "High";
    def[2] =
      "How much time pressure did you feel due to the rate of pace at which the tasks or task elements occurred? Was the pace slow and leisurely or rapid and frantic?";

    scale[3] = "Performance";
    left[3] = "Good";
    right[3] = "Poor";
    def[3] =
      "How successful do you think you were in accomplishing the goals of the task set by the experimenter (or yourself)? How satisfied were you with your performance in accomplishing these goals?";

    scale[4] = "Effort";
    left[4] = "Low";
    right[4] = "High";
    def[4] =
      "How hard did you have to work (mentally and physically) to accomplish your level of performance?";

    scale[5] = "Frustration";
    left[5] = "Low";
    right[5] = "High";
    def[5] =
      "How insecure, discouraged, irritated, stressed and annoyed versus secure, gratified, content, relaxed and complacent did you feel during the task?";

    // Pairs of factors in order in the original instructions, numbers
    // refer to the index in the scale, left, right, def arrays.
    var pair = new Array();
    pair[0] = "4 3";
    pair[1] = "2 5";
    pair[2] = "2 4";
    pair[3] = "1 5";
    pair[4] = "3 5";
    pair[5] = "1 2";
    pair[6] = "1 3";
    pair[7] = "2 0";
    pair[8] = "5 4";
    pair[9] = "3 0";
    pair[10] = "3 2";
    pair[11] = "0 4";
    pair[12] = "0 1";
    pair[13] = "4 1";
    pair[14] = "5 0";

    // Variable where the results end up
    var results_rating = new Array();
    var results_tally = new Array();
    var results_weight = new Array();
    var results_overall;

    var pair_num = 0;
    for (var i = 0; i < NUM_SCALES; i++) results_tally[i] = 0;

    // Used to randomize the pairings presented to the user
    function randOrd() {
      return Math.round(Math.random()) - 0.5;
    }

    // Make sure things are good and mixed
    for (i = 0; i < 100; i++) {
      pair.sort(randOrd);
    }

    // They click on a scale entry
    window.scaleClick = function scaleClick(index, val) {
      results_rating[index] = val;

      // Turn background color to white for all cells
      for (i = 5; i <= 100; i += 5) {
        var top = "t_" + index + "_" + i;
        var bottom = "b_" + index + "_" + i;
        document.getElementById(top).bgColor = "#FFFFFF";
        document.getElementById(bottom).bgColor = "#FFFFFF";
      }

      var top = "t_" + index + "_" + val;
      var bottom = "b_" + index + "_" + val;
      document.getElementById(top).bgColor = "#AAAAAA";
      document.getElementById(bottom).bgColor = "#AAAAAA";
    }

    // Return the HTML that produces the table for a given scale
    function getScaleHTML(index) {
      var result = "";

      // Outer table with a column for scale, column for definition
      result += "<table><tr><td>";

      // Table that generates the scale
      result += '<table class="scale">';

      // Row 1, just the name of the scale
      result +=
        '<tr><td colspan="20" class="heading">' + scale[index] + "</td></tr>";

      // Row 2, the top half of the scale increments, 20 total columns
      result += "<tr>";
      var num = 1;
      for (var i = 5; i <= 100; i += 5) {
        result +=
          '<td id="t_' +
          index +
          "_" +
          i +
          '"   class="top' +
          num +
          '" onMouseUp="scaleClick(' +
          index +
          ", " +
          i +
          ');"></td>';
        num++;
        if (num > 2) num = 1;
      }
      result += "</tr>";

      // Row 3, bottom half of the scale increments
      result += "<tr>";
      for (var i = 5; i <= 100; i += 5) {
        result +=
          '<td id="b_' +
          index +
          "_" +
          i +
          '"   class="bottom" onMouseUp="scaleClick(' +
          index +
          ", " +
          i +
          ');"></td>';
      }
      result += "</tr>";

      // Row 4, left and right of range labels
      result += "<tr>";
      result +=
        '<td colspan="10" class="left">' +
        left[index] +
        '</td><td colspan="10" class="right">' +
        right[index] +
        "</td>";
      result += "</tr></table></td>";

      // Now for the definition of the scale
      result += '<td class="def">';
      result += def[index];
      result += "</td></tr></table>";

      return result;
    }

    function onLoad() {
      // Get all the scales ready
      for (var i = 0; i < NUM_SCALES; i++) {
        document.getElementById("scale" + i).innerHTML = getScaleHTML(i);
      }
    }

    // Users want to proceed after doing the scales
    function buttonPart1() {
      // Check to be sure they click on every scale
      for (var i = 0; i < NUM_SCALES; i++) {
        if (!results_rating[i]) {
          alert("A value must be selected for every scale!");
          return false;
        }
      }

      // Bye bye part 1, hello part 2
      document.getElementById("div_part1").style.display = "none";
      document.getElementById("div_part2").style.display = "";

      return true;
    }

    // User done reading the part 2 instructions
    function buttonPart2() {
      // Bye bye part 2, hello part 3
      document.getElementById("div_part2").style.display = "none";
      document.getElementById("div_part3").style.display = "";

      // Set the labels for the buttons
      setPairLabels();
      return true;
    }

    // Set the button labels for the pairwise comparison stage
    function setPairLabels() {
      var indexes = new Array();
      indexes = pair[pair_num].split(" ");

      var pair1 = scale[indexes[0]];
      var pair2 = scale[indexes[1]];

      document.getElementById("pair1").value = pair1;
      document.getElementById("pair2").value = pair2;

      document.getElementById("pair1_def").innerHTML = def[indexes[0]];
      document.getElementById("pair2_def").innerHTML = def[indexes[1]];
    }

    // They clicked the top pair button
    function buttonPair1() {
      var indexes = new Array();
      indexes = pair[pair_num].split(" ");
      results_tally[indexes[0]]++;

      nextPair();
      return true;
    }

    // They clicked the bottom pair button
    function buttonPair2() {
      var indexes = new Array();
      indexes = pair[pair_num].split(" ");
      results_tally[indexes[1]]++;
      nextPair();
      return true;
    }

    // Compute the weights and the final score
    function calcResults() {
      results_overall = 0.0;

      for (var i = 0; i < NUM_SCALES; i++) {
        results_weight[i] = results_tally[i] / 15.0;
        results_overall += results_weight[i] * results_rating[i];
      }
    }

    // Output the table of results
    function getResultsHTML() {
      var result = "";

      result +=
        "<table><tr><td></td><td>Rating</td><td>Tally</td><td>Weight</td></tr>";
      for (var i = 0; i < NUM_SCALES; i++) {
        result += "<tr>";

        result += "<td>";
        result += scale[i];
        result += "</td>";

        result += "<td>";
        result += results_rating[i];
        result += "</td>";

        result += "<td>";
        result += results_tally[i];
        result += "</td>";

        result += "<td>";
        result += results_weight[i];
        result += "</td>";

        result += "</tr>";
      }

      result += "</table>";
      result += "<br/>";
      result += "Overall = ";
      result += results_overall;
      result += "<br/>";

      return result;
    }

    // Move to the next pair
    function nextPair() {
      pair_num++;
      if (pair_num >= 15) {
        document.getElementById("div_part3").style.display = "none";
        document.getElementById("div_part4").style.display = "";
        calcResults();
        var finalResults = { scales: scale, ratings: results_rating, tallies: results_tally, weights: results_weight }
        jsPsych.finishTrial(finalResults)
      } else {
        setPairLabels();
      }
    }


    onLoad()
  },
};
// usage
jsPsych.init({ timeline: [{ type: 'tlx'}], on_finish: function() { jsPsych.data.displayData() }})
```
Let me know if you have any further questions!

[demo]: https://flems.io/#0=N4IgtglgJlA2CmIBcAWAnAOjQVgDQgGcBjAJwHtZZkBtABl1oF18AzCBAm0AOwEMxESEBgAWAFzBV8RMtzHw5yECAC+uHv0HCAVp2mz5ioboAKBAJ5ERGAA6wArgHMI3AtQA6IMbAAenxgAEALwBwO7cAQEuLGRIoQE2vCSa8iQEccAqAWrhkWIkELywcSz23ERiELIBABRQEAR2vOYA+vAIAnK4AfmFsACUobmRAfWNsM1tHQpiGC7c8CQAEgAqALIAMsEBAAbDIweRADz1AG5RUEGeZy2JJGIAjJ4AfPuHIyu8BADWAQCK9ngBEqsj4EBI8ACAFoAiYkmIAg8AkcAEYkAIAeleEXeB1R6Kxb1xAGFYBAiL9qvBeFYAsQipDeAixCJITYyC5mSImQEUUCES56kQmUCAuYyPZ0fAfDZFhAFERIWQWD1WT0vr98Zjsbi8WjtUT3icIOdoFdCMKELQXkcMWcdbrjmcLub6QgniBnrb7YbDsbTZdPG74AAmG12k0Ox3+l1By3wADM4Z9OOjzrNcYZKGTkd9ePTgYtDOwOdOUd1WsJqYrLhs9gRRAmBAI5oWPjEnljIDbHZAPXMsvNKPrYlkndORUB5uJBhcgICz1efdkjfJ3yHI9kcPuDxq-QA3C88yNvbnq7iYxmQDc7mIw33geYEOaxk1zEhuLJ4Efz+9Pj9-kBYEqm4MEIWhWF4QCENkX1KtHWOODy1xAB5BYAmpWllVVSEYkoMgAHcXEcRFsDpUh4AUAhulXCkAmqFlIWDHoIG8SEWR5CEbAhAgZgIHDj0iMAyHAiAwHZe5eDkAIZDkAph1HdFRwCAiRO+WAyF4KAAhiJS1TEDVYIJZCjSQwTkVresZKbFtPB7Tsrx7e9+0HTwFNHbhx0neBp1nbh50XTtZFJNcNzEDztzvPdD09czTzLOKCxfE1bnhJMHzEJ8fOuBo3w-L8fwQgJ-1+AF+RAsDIRhSLoKMg1fz9MyGoOEK6IYtUWBpRTVU4+BuKBPicICYTRPEkSDOk2TegUkSejIFS1I0rSdNmxj1R+SsTMa4y4oMlEEC2vF8kOk8xCgL1LIbGzzUScEHMLW6SA9FzspAdyxz7CcHFeoLuFa9c3M3bg4XBXcDxeZEMTOk7jjO6yvls694BYG0ks8R6HhaKBkdLL0ofOuKoZIGHkWO3btKKCBHG4c1FTkRYIdm21ofJvGWeak8iZJo4yY52HzqOS74ebG7eDuvsr0e5zMtct6ga877zV+-6woisWSBDaLw3Zoqee0xsEZfHHPUvB71ZDLHjbx+1mYJvnId53Wod4fb4G5iMEvt03ktOVL7mzDKsuS8ZmnyhZCoQ+KSZ2Q9POrIjuCgQiMHe4H4SREJU8ih5Y4OBOk4IlOgcimDM+L+EQ1zkZ8+TrP1Yz3ly9BqvIhrwu6-BUvG-Crdzdjt4J3RZiQgWAiAgAQRIZJzC16tB4CBAWAREf4DHyfp9ng554KRxxG2UeJ6n5pN5GefsZVFe16PmfwbnpIAgAOQAVTWFoAGViXHjYAFE3+2AA2fu1Zgx0ECCETwawZhFACAAEXgGAKSUBPAtwXsjMQoDtieA2IRZBbwd7iAweAkASwqYiFwdWc+hDDSeCWIRYa9haSdAMrAAIiCEiLEVDYMQ9hoFdRNKxcwKkvgBAhAAR3sOCeA2kajwAwI4DAqoXDfGIt0bGRBoAqJkkUIg9gJiVG4I4boEIBBgD5AUAx3QNJkGURYuk1ISBWE0fAMQRB+gAH4AgAHVhFrQMgBakFh6LomxggxOmiCBiTsEqdEMhxIIB8N0aUfCDFBJWiQZwpxiJuPIW8EBDwwEBE8CYEQ5gImWlgfAxB5CDiL3QfkzBIBsEEWqSMfBdSCk0NIS0yIlD6lXF-DQuhYAGEiASCUspvCKj8MykI-iYiJEQmkbI+RCR7AEBEJouslBNHcJINwTRU1yDbNsXwic+jDEYRce4rxPj9KGQCYI2aITEHhI0mPWaaIGjfG6AQCYbUh75AUPYCU1ERH8lKCw2aEwUQiSqGs7JIAgEHBASGDpIAVjwIktAuBoSkGIreLU6gqKGlNO6SI0h6DiVEJIbvMllCqXUOIUMkZLEBAJB4gQSUkJ6jaXFPYHSlEWFQHnMpNayR5D0RVIkRUrCEQEQ2bSXxGp+KrXudMOQKqiA6KnlIjx3j+JrWlUxN5rDE6oIaFy2Ajz0TJBsNAU12kWDJDkOSBFSKRggITGikwixdKhMVGSwlXqGkAHEyBkDxSgtp1Bg1EJMOGkgdLkYxrAYy2hY9OVaqBAQCFox5p8sUdwX4BaCKLEhC4VhWqyBxIaBslJa1HCaVgCqlUSqAK8QRCiQRa1pSygKEwxYtRZp8rSO0Fg1z010iZA0NgUiVJlrFBKFSrFRkjvYSQP1UkZUVppLEuwtbiI4V4gERtRQCBuvCLk+M1AUBou-iwXSvYUGEtvSSnB+LqzRtfdSrpH6DiUO-Wmuh3ISDaR5Yu-l3JTjsXmqpEgvwahMKKFah1YzSnkmQ+YQYyld3Vv3esiD6IEDQchVK31Il-XwAvXHZF17sBooAGIkDWfkadH1n1oOoPRt9zS-2tIpVxtFNKyF8Z6cm7j-T3iDLHi4Xi2r4CqIaDISUvBHBSO6OCAoBl5BQB+YC5sc62FSU-OYOd0G0hrLsfJ7ojhxUQFnbpmSBgZhGPaLwHwhmzV7r+TMUY9qC0sEFaMSUh623fGo28DEGJILghbTpLqIl+IVpEtjdEFa1qwucHwFhsn8gMJBK4bo-lTGLAIJF6LEJAtKXmmtQU0oogRDWsGSxaCjEUtUcjVh18CAYAHvfR6+9V6Hw3rfA4j1CGFJACgAI6UUHjb6ZNmCJZRMJHVkStFMEA5zbW7GybSJlvbfBDetFCYAgHbeONiTe3oJkvG--NFSJZsXbWwAdg2wEa0K3xsAA40VkS289o7aATsfdu2th4TAGmnfvIdkg1AHgLc8LQAIAPqzzapZN5HHpYfw9254ab2PAdw4eN+ybZFPvusiFFgIAA1JIhRXYqVZOBMVQJdFiH4gobS9gbB9ZtWz2AHOWh2ZSZfYbx9Run3vjxdnBAWjMJQ2L9eEuUHbwF0L0tFLBtXxG6r6X6u5dkHM8hynAR56PRaMV7YtAUG6VqPPCA1v9xRGRI-F+79P4-zfs7iAABqX3gwZeC7lwr8w1AIAFJt5e6s1On68W0spZ1SdIAAC92JqkesRfi-VeL0wTzVtUayGbVlKOUArIjEEoVA3uIYv4IS7IiGsJk1hyBlCgDUJvLIMBJ+rXuQYMJaAYGwCglQ0eDjU6b98JiXLC2OH4kkSEjaI2ocgB5pBJfZo1EdyEG3LujiIloLvv3Afa-vEehgAg40ahJ6r1ASXkRR80ZGNTlYrJBG0UpBEXg5EGQYTkuYN4NuC-eMf6bYUvKZaoYMf6LfROaUboL6QYMIOvA3YXadAxcPWAnwApL6U3KnaLFYSUCIFEGkb4WzCULzCgVaWDDZCVO3ZDGSdoZtQ0O3LfbYYfPfEICHI-AIX3EIbARAvMeeUcGwBpMQFoTsX3BrbGHwHgybcQvsSQiAFBLee+GFHuMABpFEeQ2QurGQyQzwbQxQ5QkYJOHRJhORZxb+dVMQAAIXMAAEl29hD+gU5HAZwNJ0QiEABiBjXwvwslf9MgMwmYCwsQKw+BGYOwxwmoNQ0cMAFwlENwygzwybHwvw3wgI7IMfQ4IQsgEQohMQiQqQ+rfQkAQws3IoYw+eWI6tTQ8o3Q2QgwoonAw0Uw+wcwtTMI6wqIpwvIhIpIjwhpLw8eEY0YzItojoyw7ohw9vGo+I1w9w2abw0YsYlbR-crAIAAJWcUIKGlWE2B6gRG4gjQYVFCVUZzoJPRNAUB-wQDeHAPL06LfnjH2I2BgOkIEN-DV05UFwaRyV-GpxQnrEHT2gQGXRZFYScwcDAAiDt2ayhPaNhKeWRhcFYhAkNCDwRF4Mmx5hdgOh5mJj1iXFwMxHwLxPTx5DUwWHFTOLVGDAxPVx4JCAAHJcTGcDYRZMw7jPRmSSTqdNi6EHhuhtAWMho+A2VsImt4wGSfisTJNcRWTjo9YoTGgpJzQQxPthZEZWQtJiIXhmTZCQFdDAhSjbY2ZiZ-j3h+S6EQxuhfE8iAhuRYAVRJS6T4wGsKImFQUNS5pmEESYSytkDZSmScSyYVtIh54rdODjCWCHc2CfdkRODD8EzsT+DT9cRMSmTBJFTtIrwxCDTfdBIGjCy+YmiFCiyeDszOxIgOTEZhCCzBIrcSyEJmTfo1gQV4An4bBXQQCyQKQagGy+ZizBJPBugJCKzmzHRmTwYTZ8ZnheS8xit-djDIh7NagrdnhoJBgozERjDH93hMzsTPBmYLSP1DRrSx4Exug5jHSigXTW03Tf8XBPS+IZT2cQzjywyYzN84y+CEyjgkzuDUzPjdRDz5TdQcyuwtDByEJhzSyyjxyhzKy+ZWy+wazro3IyB1C2yOyuyeyGRoCYLHQ4KEJRzJtkLYKKKpyZy2ZzoFzfx9zDgwKcSuZLTDgLyUcWsl5UM2lJUK8DFIRoV2hAyDzGSjyQACTiS3zfjeC8xIKZBm1EgaZPAIdOxazzRal9SqL3hCVjTtLDhWS5zlSFLVTlKQBVK+x1LPA2ktLJyDyBM9K7LDhjy5zMjmKXKlTnZXZaKpKn9x9osH46E7c1pz5UTy9XSmJpSgz3zsT5KMKkYUYeTjDmLKFjTkqxKiEzSTy2ZyTfKGSG8wVZSR9sjIgHiQJ6JuBsEtIa8kCrTosQ1nFWFKAhpgw5lqQoAADfxYz75t8Pt-y3dX4P4v5f4fd-cQLcQJiQjOjwimEeiaguTvwFCogXD5hFhXjtgniXj1g3iIB78RhGKH8SrSSAg49SshFpJlJjjFRDMl5B0k5QtHyOB7iygICiDy4dwarzzotiRWQ6JlI+Q6QZ9GJ38+zP8MJzNBF6Suqfyeqnc98BqPdhrvcogxr0z3g1yagABCTEuXEXRwcPRgca3UBke4eakAceCo76ehYEXkJidoeACoOdO3eACG24+ATGzwPa0SgqzqZteAFcrIhio6vAgIOwyELtNkKCIUx0xg+aW8G7X8SauQUImayImYsmm8dOTmi-TKBADAV8CYQRIhT8cOcMvNYI5W6a6Y6InKX2W8e8Fwx8PWg25oP4s85AgqvLfmt4dYmPaLU64JL8MFXU+tDPKCGCXLZjV6kSkYMq6oOue4TWIm5-aLMW3kUzVbe4aCboVkfCTOhEJMRWoI9oqaqYiIuQOa22v2O8bWp22RF2o2ybE2xa4wpW2YK28u2w9Wqu28dKR23W+u3KQ2t2vk6LN+RqtaIS5tNJIaVOGO5FZxEGEgDYF2YSk+A4evXYr24qvy468erkcWoGBeVe6e4KsO8EIiY9PdenS-CIYEVTb8EvF68vDtJelevkZtT6r4nqzA0UJXa+dekYXQv+1bI7C3YrRgC-fdMQMmybW+Q0c3eubYI03+twJgRgKo-rc2ZB69YBtwfJDB4W82kuy2su2a7ukADGbWr6ecEIDGVu4uyYrozuyuyh82ah7ybYKWEktulW629vdGeuS2RKla0CNa7a7YVK1B0BQhoui29ushtWm2thzuYR7W1a5YCRkIKR6QoEeHRgWRg4X2-y4qN-ayNcOde0kQgbVOZ6svcqjuJ6L+94B3VB7XcXG+YwvBrhtbcB9oyB8YViWBzm9K2UkPTDDA3RtBgx5cohnsJewByITevZHoZjb26sYxlO0x+AEGixhPNUG8mx+WJ++x+OpuDWZxnIn+qJ9x5XTxw0bxuh3x9WS3fxqBskGBzsEJ6SoXUPSJ6UPRgh2J38eJ9WRJsFT2tJnejYmccSYEoaTXXeDnVDNaNgbLciESR+g4OOiIS0IgbYsJypjelAo3RYegnfDAKPXe0qmGtLOGx3ffZ+Qaz3EalGk-WqjMlAxZghCPbYHG+XCJ356LB4bAS5gW-5055IZq7E-5759BX5gAKkKtl1QPOQJr3J9qOsBPrDrAPvVEZ2whxrsdepPWcQOdl1eKOal352DKITYuOeDNkuak-NyskqJKyuhk2LQMcA5fOk+EoHMF5eeE8XgApSytYrNu6ruZ336qecRq91GvebzHcokq-KIaSYytDIJjNo1cZZCCNMMc+b1ZYpZndtAs1c-O1fBc1f+bxvReVYtYktcrNaNZisytNeteNf+b6Yj09bdZNatfVeRZkvdcDeamYthdFaWftfDcdbNPpaYrjYlYxeueDblJNdyrcrjbREJDNpVZQmN2aqIVCf9YhcLdgBLZDZxJzbyo9t2MxOmb9oCHbOgzmjFOlCOPVmJfL1GfBCpciD8bAGGfRpVBqEHYXE4LTI+feF4Y7vIeUc1vuD7p1qyn1qHtduNoKh1eIaYdVoroocXbEADn7tXYbpHrzD2fJeD1nkEPvjWaKCveWZCGADZvSDZqMW5bfdte5btOQ3lC-ZQND26DhYA7CZaDhcFt1FMAsCsAwDWdrRWAKCKBqHvdgEfYIH6ENCyGEshGncOFfvVnfrXq5sg-2ujzeFkCqvb0w5xByG4BUCAWg8sGsDCpqBfcqAEDJAWDiGoHY4HHgDiGZO8B8GZJUGYAqpaHg-WRKGfpAhqoCCY9g6gCZF4DXZDnMBgRU5rxUBUBo5AHwF4gQGjqUATGwDQCQFoFUHUG7E0CUAwCIGbH05ACmhmCUDOhTiwriPCCQPcDECIigBZDiEH3-iIDAGMNZApSC4wBQFC-7jEBhVA0WChDmLiAeBsBkMvzJG0n2hILi4S9SyhFqVS-S7pAoHtRy4pDy5SyS7aWK4y7K+y7+W+Di4QXSRcCC-S+d0vTEESBgGIg65kNoBK6G8G86+j19tBNkShp84kCSCywG7i96-qAMQG4+2G-W58GMPy6S4UomBsF4jiF294H28fom6gAwGEKeG4Bm-88C4+wwBC7C8NAi6Wai5i6e88ni+q5IChGELq9K6y95Ca6q8S5+6K8RBK8y-K+B+69a-m7W828KU+6W-64R7R5G4R9jkf3c+ELDGu+69u5ECi8e-C6jfEDe9i+6+25+7+4h-q8B4q+a6p++6hFq7p4B+h9y9h7m-a8x6R985R5W-R428x-CGx-O51OW8cG8+65iDkDiBsIoG0hJxK8nj6G6CWHaGg0qGFB+SkgIChF4gKBYDi-kHbChEpmpkO5mEWCx-CHc80vx8+7l7EFS5QFV6Q9gA161+cQwz19cEN7lBN7F-t-O5sqd985d7d49-V4CE19gG1796nQD6N-s1N47Yt7JCt-JSWbt7jnc-Phl8+8J9S5DEp47GR60il9W4x4x41LG+d4MFL5j6KG94T9991+T4N9T+D-o+j0ulsC7Yj78+gDu+wEp8+5e-J8RAn8j6b9FqV8RHd5kLV9b7j595194H9+76D6x6c8M4ZoK04CEBQBeyQAeFUGYBAC4++GP945s4ECUAhD4VkF+7EngChF0H2+Y4wHaPO90Cc6SgqAQgcQD13SBRYygNgUgvZ2rQYgAAApkm0DNBWuPOewPUAxDP9Xqb-AQHAMHyD4IcEYYEBiGHCJwOAGAzCAVmwEf8v+MHawH-wwAAD8AMsLQMQAKBcJL+KgIAA

**Reply by jenniekim00** [reply]
Thank you so much, it worked like a charm! I am still trying to decipher and understand it myself, but it's definitely working and I can't thank you enough for your help. You are a wonderful soul :'))

**Reply by vijaymarupudi** [reply]
Feel free to ask questions if you come across roadblocks!
